src/function/audio.rs

```diff
@@ -0,0 +1,296 @@
+            self.dir.write("p_volume_present", (volume_present as u8).to_string())?;
+        }
+        if let Some(volume_min) = self.builder.playback.volume_min {
+            self.dir.write("p_volume_min", volume_min.to_string())?;
+        }
+        if let Some(volume_max) = self.builder.playback.volume_max {
+            self.dir.write("p_volume_max", volume_max.to_string())?;
+        }
+        if let Some(volume_resolution) = self.builder.playback.volume_resolution {
+            self.dir.write("p_volume_res", volume_resolution.to_string())?;
+        }
+        if let Some(volume_name) = &self.builder.playback.volume_name {
+            self.dir.write("p_fu_vol_name", volume_name)?;
+        }
+        if let Some(terminal_type) = self.builder.playback.terminal_type {
+            self.dir.write("p_terminal_type", terminal_type.to_string())?;
+        }
+        if let Some(input_terminal_name) = &self.builder.playback.input_terminal_name {
+            self.dir.write("p_it_name", input_terminal_name)?;
+        }
+        if let Some(input_terminal_channel_name) = &self.builder.playback.input_terminal_channel_name {
+            self.dir.write("p_it_ch_name", input_terminal_channel_name)?;
+        }
+        if let Some(output_terminal_name) = &self.builder.playback.output_terminal_name {
+            self.dir.write("p_ot_name", output_terminal_name)?;
+        }
+
+        // general
+        if let Some(fb_max) = self.builder.fb_max {
+            self.dir.write("fb_max", fb_max.to_string())?;
+        }
+        if let Some(request_number) = self.builder.request_number {
+            self.dir.write("req_number", request_number.to_string())?;
+        }
+        if let Some(function_name) = &self.builder.function_name {
+            self.dir.write("function_name", function_name)?;
+        }
+        if let Some(control_name) = &self.builder.control_name {
+            self.dir.write("if_ctrl_name", control_name)?;
+        }
+        if let Some(clock_source_in_name) = &self.builder.clock_source_in_name {
+            self.dir.write("clksrc_in_name", clock_source_in_name)?;
+        }
+        if let Some(clock_source_out_name) = &self.builder.clock_source_out_name {
+            self.dir.write("clksrc_out_name", clock_source_out_name)?;
+        }
+
+        Ok(())
+    }
+}
+
+/// USB Audio Class 2 (UAC2) function.
+#[derive(Debug)]
+pub struct Uac2 {
+    dir: FunctionDir,
+}
+
+impl Uac2 {
+    /// Creates a new USB Audio Class 2 (UAC2) builder with g_uac2 audio defaults.
+    pub fn builder() -> Uac2Builder {
+        Uac2Builder { capture: Uac2Config::default(), playback: Uac2Config::default(), ..Uac2Builder::default() }
+    }
+
+    /// Creates a new USB Audio Class 2 (UAC2) function with the specified capture and playback channel mask, sample rate (Hz), and sample size (bytes).
+    pub fn new(capture: (u32, u32, u32), playback: (u32, u32, u32)) -> Uac2Builder {
```

Comment:
Could you define a struct `Channel` to make the API a bit more accessible?
This would allow rust-analyzer to provide completions to IDEs.

```suggestion
    pub fn new(capture: Channel, playback: Channel) -> Uac2Builder {
```

and 

```rust
pub struct Channel {
...
}

impl Channel {
   pub fn new(channel_mask: .., sample_rate: .., sample_size: ..) -> Self { ... }
}